### DISCUSSION TITLE
solution/20230626B_D/

### DISCUSSION THREAD

**Discussion by giscus** [X-six-floor-slip-liu-X/X-six-floor-slip-liu-X.github.io#3] [Announcements]
# solution/20230626B_D/

这是一个中学 OIer 的博客，他会在这里分享一些算法和题解，喜欢可以进来贴贴。

https://x-six-floor-slip-liu-x.github.io/solution/20230626B_D/

**Comment by Enonya** [comment]
怎么想到先不考虑进行操作 2 的。

**Reply by Enonya** [reply]
可以注意到这个 dp 方程的本质其实建立在，我们“假定”操作 2 都已经完成，剩下的节点不满足的一定能一次满足的前提下。
我们实际上是把操作 2 的整体影响拆成子树外和字数内两部分以达到“都完成”的目的，那为什么需要先进行操作 1.

**Reply by X-six-floor-slip-liu-X** [reply]
~~这你得问凯伊斯特·欧埃尔呀~~
